tests/agent_test.py

```diff
@@ -268,3 +290,27 @@ def testTsunamiAgent_WhenMessageIsIpRange_ShouldCallTsunamiForAllHosts(mocker: p
     msg = message.Message.from_data(selector='v3.asset.ip.v4', data={'version': 4, 'host': '0.0.0.0', 'mask': '28'})
     tsunami_agent_no_scope.process(msg)
     assert tsunami_scan_mocker.call_count == 14
+
+
+def testAgentTsunami_whenIpRangeScanned_emitsExactIpWhereVulnWasFound(
+    ip_small_range_message: message.Message,
+    tsunami_agent_no_scope: ts_agt.AgentTsunami,
+    agent_mock: List[message.Message],
+    mocker: plugin.MockerFixture
+) -> None:
+    data = {
+        'scanStatus': 'SUCCEEDED',
+        'vulnerabilities': [
+            {
+                'vulnerability': {
+                    'title': 'Ostorlab Platform',
+                    'description': 'Ostorlab is not password protected'
+                }
+            }
+        ]
+    }
+    mocker.patch('agent.tsunami.tsunami.Tsunami.scan', return_value=data)
+    tsunami_agent_no_scope.process(ip_small_range_message)
+
+    assert 'v3.report.vulnerability' in [a.selector for a in agent_mock]
+    assert ['ipv4'] in [list(a.data.get('vulnerability_location', {}).keys()) for a in agent_mock]
```

Comment:
Just prepare the full vuln location dict and check if it is part of the sent data.